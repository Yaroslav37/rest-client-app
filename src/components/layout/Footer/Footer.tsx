'use client';

import Image from 'next/image';

const GITHUB_ICON = {
  src: '/github_icon.svg',
  width: 16,
  height: 16,
  className: 'h-4 w-4',
};

const TEAM_LINKS = [
  {
    name: 'Yaraslau Minenkou',
    href: 'https://github.com/Yaroslav37',
    icon: GITHUB_ICON,
  },
  {
    name: 'Veranika Liauchuk',
    href: 'https://github.com/adAstra97',
    icon: GITHUB_ICON,
  },
  {
    name: 'Elena Ivanova',
    href: 'https://github.com/Elena-lucky',
    icon: GITHUB_ICON,
  },
];

const RSS_ICON = {
  src: '/rss-logo.svg',
  width: 32,
  height: 32,
  className: 'h-8 w-8',
};

export default function Footer() {
  return (
    <footer className="flex flex-wrap flex-col items-center justify-center pb-10 gap-2 border-t border-light-green bg-dark-grey py-4 text-sm xs:gap-4 sm:flex-row sm:gap-16">
      <div className="flex flex-col gap-2 mds:gap-20 xs:flex-row">
        {TEAM_LINKS.map((member) => (
          <a
            key={member.href}
            href={member.href}
            target="_blank"
            rel="noopener noreferrer"
            className="flex items-center gap-2 transition-colors text-white hover:text-light-green"
            aria-label={`${member.name} GitHub profile`}
          >
            <Image
              src={member.icon.src}
              alt=""
              width={member.icon.width}
              height={member.icon.height}
              className={member.icon.className}
              aria-hidden
            />
            {member.name}
          </a>
        ))}
      </div>

      <span className="text-light-green">2025</span>

      <a
        href="https://rs.school/courses/reactjs"
        target="_blank"
        rel="noopener noreferrer"
        aria-label="RS School React course"
        className="transition-opacity hover:opacity-80"
      >
        <Image
          src={RSS_ICON.src}
          alt="RS School logo"
          width={RSS_ICON.width}
          height={RSS_ICON.height}
          className={RSS_ICON.className}
        />
      </a>
    </footer>
  );
}
