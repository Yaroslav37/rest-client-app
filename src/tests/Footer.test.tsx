import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { describe, expect, it } from 'vitest';

import Footer from '@/components/layout/Footer/Footer';

vi.mock('next/image', () => ({
  default: (props: any) => <img {...props} />,
}));

describe('Footer Component', () => {
  it('renders correctly with all team members', () => {
    render(<Footer />);

    const teamLinks = ['Yaraslau Minenkou', 'Veranika Liauchuk', 'Elena Ivanova'];

    teamLinks.forEach((name) => {
      expect(screen.getByText(name)).toBeInTheDocument();
    });
  });

  it('contains correct GitHub links for each team member', () => {
    render(<Footer />);

    const memberLinks = [
      { name: 'Yaraslau Minenkou', href: 'https://github.com/Yaroslav37' },
      { name: 'Veranika Liauchuk', href: 'https://github.com/adAstra97' },
      { name: 'Elena Ivanova', href: 'https://github.com/Elena-lucky' },
    ];

    memberLinks.forEach((member) => {
      const link = screen.getByText(member.name).closest('a');
      expect(link).toHaveAttribute('href', member.href);
      expect(link).toHaveAttribute('target', '_blank');
      expect(link).toHaveAttribute('rel', 'noopener noreferrer');
    });
  });

  it('displays GitHub icons for each team member', () => {
    render(<Footer />);

    const githubIcons = screen.getAllByAltText('');
    expect(githubIcons).toHaveLength(3);

    githubIcons.forEach((icon) => {
      expect(icon).toHaveAttribute('src', '/github_icon.svg');
      expect(icon).toHaveClass('h-4 w-4');
    });
  });

  it('shows the current year', () => {
    render(<Footer />);
    const currentYear = new Date().getFullYear().toString();
    expect(screen.getByText(currentYear)).toBeInTheDocument();
  });

  it('contains RS School link with correct attributes', () => {
    render(<Footer />);

    const rsLink = screen.getByLabelText('RS School React course');
    expect(rsLink).toHaveAttribute('href', 'https://rs.school/courses/reactjs');
    expect(rsLink).toHaveAttribute('target', '_blank');
    expect(rsLink).toHaveAttribute('rel', 'noopener noreferrer');
  });

  it('displays RS School logo with correct attributes', () => {
    render(<Footer />);

    const rsLogo = screen.getByAltText('RS School logo');
    expect(rsLogo).toHaveAttribute('src', '/rss-logo.svg');
    expect(rsLogo).toHaveClass('h-8 w-8');
  });

  it('has correct styling classes', () => {
    const { container } = render(<Footer />);
    const footer = container.firstChild;

    expect(footer).toHaveClass('flex');
    expect(footer).toHaveClass('flex-col');
    expect(footer).toHaveClass('items-center');
    expect(footer).toHaveClass('justify-center');
    expect(footer).toHaveClass('border-t');
    expect(footer).toHaveClass('border-light-green');
    expect(footer).toHaveClass('bg-dark-grey');
  });

  it('applies hover styles to team member links', async () => {
    const user = userEvent.setup();
    render(<Footer />);

    const firstMemberLink = screen.getByText('Yaraslau Minenkou').closest('a');
    expect(firstMemberLink).toHaveClass('text-white');
    expect(firstMemberLink).toHaveClass('hover:text-light-green');

    await user.hover(firstMemberLink!);
  });

  it('applies hover styles to RS School link', async () => {
    const user = userEvent.setup();
    render(<Footer />);

    const rsLink = screen.getByLabelText('RS School React course');
    expect(rsLink).toHaveClass('hover:opacity-80');

    await user.hover(rsLink);
  });

  it('matches snapshot', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
