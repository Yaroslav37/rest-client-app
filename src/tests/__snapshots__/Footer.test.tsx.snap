// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > matches snapshot 1`] = `
<div>
  <footer
    class="flex flex-wrap flex-col items-center justify-center pb-10 gap-2 border-t border-light-green bg-dark-grey py-4 text-sm xs:gap-4 sm:flex-row sm:gap-16"
  >
    <div
      class="flex flex-col gap-2 mds:gap-20 xs:flex-row"
    >
      <a
        aria-label="Yaraslau Minenkou GitHub profile"
        class="flex items-center gap-2 transition-colors text-white hover:text-light-green"
        href="https://github.com/Yaroslav37"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt=""
          aria-hidden="true"
          class="h-4 w-4"
          height="16"
          src="/github_icon.svg"
          width="16"
        />
        Yaraslau Minenkou
      </a>
      <a
        aria-label="Veranika Liauchuk GitHub profile"
        class="flex items-center gap-2 transition-colors text-white hover:text-light-green"
        href="https://github.com/adAstra97"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt=""
          aria-hidden="true"
          class="h-4 w-4"
          height="16"
          src="/github_icon.svg"
          width="16"
        />
        Veranika Liauchuk
      </a>
      <a
        aria-label="Elena Ivanova GitHub profile"
        class="flex items-center gap-2 transition-colors text-white hover:text-light-green"
        href="https://github.com/Elena-lucky"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt=""
          aria-hidden="true"
          class="h-4 w-4"
          height="16"
          src="/github_icon.svg"
          width="16"
        />
        Elena Ivanova
      </a>
    </div>
    <span
      class="text-light-green"
    >
      2025
    </span>
    <a
      aria-label="RS School React course"
      class="transition-opacity hover:opacity-80"
      href="https://rs.school/courses/reactjs"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="RS School logo"
        class="h-8 w-8"
        height="32"
        src="/rss-logo.svg"
        width="32"
      />
    </a>
  </footer>
</div>
`;
